qutip/qip/pulse.py

```diff
@@ -418,6 +418,8 @@ def get_full_tlist(self):
         if not all_tlists:
             return None
         full_tlist = np.unique(np.sort(np.hstack(all_tlists)))
+        diff = np.append(True, np.diff(full_tlist))
```

Comment:
I forget why I put True here... The purpose is to make sure that it is larger than the `tol` so that the first element is always kept. I changed it to `10 * tol` now.

Reply:
Yeah, much better to have an explicit number that's larger than `tol`.  In theory, using `True` won't always succeed - if `tol` is larger than 1, for example.  The call still feels rather strange to me - the signature is meant to be `np.append(array, values)`, so it seems almost a bug that the call works with the array being a scalar! - but that may be preferable to the alternative (avoiding the sentinel value)
```python
full_tlist = np.unique(...)
full_tlist = np.concatenate((full_tlist[:1], full_tlist[1:][np.diff(full_tlist) > tol]))
```

Reply:
Yeah, `np.append` here does seems to be a strange usage. I guess NumPy automatically promotes the first value to an array. To prevent future NumPy updates from breaking this, yours is safer.

Update made.